CONTRIBUTING.md

```diff
@@ -9,112 +9,141 @@ There are 4 main ways of contributing to the PyMC project (in descending order o
 * Contributing or improving the documentation (`docs`) or examples (`pymc/examples`)
 * Submitting issues related to bugs or desired enhancements
 
-# Opening issues
+## Opening issues
 
 We appreciate being notified of problems with the existing PyMC code. We prefer that issues be filed the on [Github Issue Tracker](https://github.com/pymc-devs/pymc/issues), rather than on social media or by direct email to the developers.
 
 Please verify that your issue is not being currently addressed by other issues or pull requests by using the GitHub search tool to look for key words in the project issue tracker.
 
 Filter on the ["beginner friendly"](https://github.com/pymc-devs/pymc/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22) label for issues which are good for new contributors.
 
-# Contributing code via pull requests
+## Etiquiette for code contributions
+* When you start working working on an issue, open a `Draft` pull request as soon as you're doing the first steps (see steps below).
+* Before opening a PR with a new feature, please propose it in an [Issue](https://github.com/pymc-devs/pymc/issues) or [Discussion](https://github.com/pymc-devs/pymc/discussions) with the maintainers. Depending on the proposal we might direct you to other places such as [`pymc-experimental`](https://github.com/pymc-devs/pymc-experimental) or [`pymc-examples`](https://github.com/pymc-devs/pymc-examples).
+* Any issue without an open pull request is available for work.
+   * If a pull request has no recent activity it may be closed, or taken over by someone else.
+   * The specific timeframe for "recent" is hard to define as it depends on the contributor the specific code change, and other contextual factors. As a rule of thumb in a normal pull request with no other blockers there is typically activity every couple of days.
+   * The core devs will make their best judgement when opting to close PRs.
+* In accordance with the previous point, please don't make unrealistic "_I want to work on this issue_" pledges.
+* It's okay if priorities change, but please communicate if you can't get a PR to the finish line, or that you're taking a break. Depending on the change (urgent bugfix vs. new feature) others might want to take over to get the work done.
```

Comment:
```suggestion
* It's okay if you are delayed or need to take a break, but please leave a  comment in the pull request if you cannot get it to a state where it can be merged. Depending on the change (urgent bugfix vs. new feature) the core devs can determine if the PR needs to be reassigned for completion.
```